### PR TITLE
DEPR: Period[B]

### LINKED ISSUE #53446: DEPR: Period[B]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
NA
```

### Issue Description

TL;DR: can we deprecate BDay Period(|Dtype|Index|Array) and tell users to use a DatetimeIndex with freq="B" instead?

Context: lots of user confusion has resulted from a) using "freq" for PeriodX to mean something different than it does for DatetimeIndex (#51256, #38885, #47227) and b) using DateOffsets to define PeriodX (#5091, #13871, #38914, #38859).

The option I'm leaning toward to address this is to change/deprecate PeriodX.freq to "unit" (the attribute, the constructor keywords, and asfreq->as_unit).  The sticking point I'm finding is in finding a replacement for the current usage `period + period.freq * N`.  For everything except BDay, we could tell users to do something like `period + np.timedelta64(N, period.unit)` instead (actually just `period + N` works, but I'd actually prefer to deprecate that).

So I'm wondering if we really need to support Period[B].  Anyone have a strong opinion?

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #53446 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

There's a place in the plotting code where we apparently cast to Period, not sure why that would be necessary.  Need to find an alternative before the deprecation is enforced.

### CODE REVIEW

File: pandas/tests/plotting/test_datetimelike.py [hunk 1]
```diff
@@ -217,6 +217,7 @@ def test_line_plot_period_mlt_frame(self, frqncy):
         freq = df.index.asfreq(df.index.freq.rule_code).freq
         _check_plot_works(df.plot, freq)
 
+    @pytest.mark.filterwarnings(r"ignore:PeriodDtype\[B\] is deprecated:FutureWarning")
```
Comment (MarcoGorelli) [inline pandas/tests/plotting/test_datetimelike.py:220]: do we need to find a workaround in plotting before doing this?

Comment (jbrockmendel) [reply pandas/tests/plotting/test_datetimelike.py:220]: yes.  suppressing the warning on this test is OK, but i think we need a better solution than suppressing here https://github.com/pandas-dev/pandas/pull/53511/files#diff-9e7653bee8c4fb4e42459af383020b7d754df6c55e79680d20e81d588d51f029R283

**Comment by mroeschke** [comment]
+1 on deprecating. What about the other Business offsets?

**Comment by jbrockmendel** [comment]
> What about the other Business offsets?

The other business offsets aren't used for Period.  (Confusion about that is common, which is part of why i want to decouple offsets from Period entirely)

File: pandas/_testing/__init__.py [hunk 2]
```diff
@@ -441,7 +442,10 @@ def makeTimedeltaIndex(
 
 def makePeriodIndex(k: int = 10, name=None, **kwargs) -> PeriodIndex:
     dt = datetime(2000, 1, 1)
-    return pd.period_range(start=dt, periods=k, freq="B", name=name, **kwargs)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        pi = pd.period_range(start=dt, periods=k, freq="B", name=name, **kwargs)
```
Comment (mroeschke) [inline pandas/_testing/__init__.py:447]: Is it possible to use another freq like "D"?

Comment (jbrockmendel) [reply pandas/_testing/__init__.py:447]: Doing that breaks a handful of tests.  I could go either way on changing it now vs after enforcement.

Comment (mroeschke) [reply pandas/_testing/__init__.py:447]: Would slightly prefer changing now to hopefully prevent potentially having to change more tests in the future

Comment (jbrockmendel) [reply pandas/_testing/__init__.py:447]: Fair enough, updated

File: pandas/tests/resample/conftest.py [hunk 3]
```diff
@@ -63,7 +64,10 @@ def simple_period_range_series():
     """
 
     def _simple_period_range_series(start, end, freq="D"):
-        rng = period_range(start, end, freq=freq)
+        with warnings.catch_warnings():
+            # suppress Period[B] deprecation warning
+            warnings.simplefilter("ignore")
```
Comment (MarcoGorelli) [inline pandas/tests/resample/conftest.py:69]: could we be more precise with the `simplefilter` please? I think this may set a bad example for new contributors wanting to silence warnings and looking for existing patterns of how to do it

File: pandas/tests/resample/test_base.py [hunk 4]
```diff
@@ -277,15 +277,20 @@ def test_resample_size_empty_dataframe(freq, empty_frame_dti):
...
+@pytest.mark.filterwarnings(r"ignore:PeriodDtype\[B\] is deprecated:FutureWarning")
 @pytest.mark.parametrize("index", tm.all_timeseries_index_generator(0))
 @pytest.mark.parametrize("dtype", [float, int, object, "datetime64[ns]"])
 def test_resample_empty_dtypes(index, dtype, resample_method):
     # Empty series were sometimes causing a segfault (for the functions
     # with Cython bounds-checking disabled) or an IndexError.  We just run
     # them to ensure they no longer do.  (GH #10228)
+    if isinstance(index, PeriodIndex):
+        # GH#53511
+        index = PeriodIndex([], freq="B", name=index.name)
```
Comment (MarcoGorelli) [inline pandas/tests/resample/test_base.py:289]: why is this necessary? it looks like, if `index` is an empty `PeriodIndex` with `freq` 'B'`, then you're re-creating it?

File: pandas/tests/plotting/test_datetimelike.py [hunk 5]
```diff
@@ -319,11 +320,16 @@ def test_irregular_datetime64_repr_bug(self):
 
     def test_business_freq(self):
         bts = tm.makePeriodSeries()
+        msg = r"PeriodDtype\[B\] is deprecated"
+        dt = bts.index[0].to_timestamp()
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            bts.index = period_range(start=dt, periods=len(bts), freq="B")
```
Comment (MarcoGorelli) [inline pandas/tests/plotting/test_datetimelike.py:326]: why is this necessary? why are you re-creating the PeriodIndex?

**Review by MarcoGorelli — COMMENTED** [review]
generally looks good, just got a couple of questions on some tests

File: pandas/tests/resample/conftest.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/resample/conftest.py:69]: good point, will update

File: pandas/tests/resample/test_base.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/tests/resample/test_base.py:289]: per matt's request i changed the fixture, so now index.freq=="D"

**Review by MarcoGorelli — APPROVED** [review]
no objections

**Review by mroeschke — APPROVED** [review]
